### PR TITLE
general scraping improvements

### PR DESCRIPTION
**Pull request by DaniD3v**
added 3 new extractors, removed some duplicated code

### CODE REVIEW

**Comment by SunglassJerry** [comment]
What's filelions and streamwish?

**Comment by SunglassJerry** [comment]
+ what's your discord? You in the server?

**Comment by DaniD3v** [comment]
> 
Gogo has them as sources.

> +what's your discord? You in the server?
nichtdu

File: build.gradle
Comment (brahmkshatriya) [inline build.gradle:1]: Remove changes to build gradle, and make a separate pr, if you want to update them

File: app/src/main/java/ani/saikou/parsers/anime/AniWave.kt
Comment (brahmkshatriya) [inline app/src/main/java/ani/saikou/parsers/anime/AniWave.kt:1]: Revert this change, the original method is much more reliable then parsing the domain name, And I dont recommend using Aniwave Extractor outside of Aniwave

File: app/src/main/java/ani/saikou/parsers/AnimeParser.kt [hunk 1]
```diff
@@ -62,7 +82,36 @@ abstract class AnimeParser : BaseParser() {
...
+            "dood.wf"                     -> DoodStream(server)
+            "ok.ru"                       -> OkRu(server)
+
+            //currently down so I can't add the extractors
+            "mcloud.to"         -> AniWave.Extractor(server) // https://mcloud.to/e/qp2405?t=4xnZC%2FwjB1MKyw%3D% I don't think the AniWave.Extractor() is a good solution because it's site dependent.
+            "streamlare.com"    -> null // streamlare.com/e/vJ41zYN1aQblwA3g
+
+            "goone.pro"         -> null // Gogo links to this other torrent site. This could be a parsers.anime but it definitely shouldn't be an extractor.
+            else                -> {
+                Log.println(Log.INFO, "AnimeParser: $name", "No extractor found for: $domain | ${server.embed.url}")
```
Comment (brahmkshatriya) [inline app/src/main/java/ani/saikou/parsers/AnimeParser.kt:108]: use `println` instead of `Log.println`, cause please keep logging consistent 😭

File: app/src/main/java/ani/saikou/parsers/AnimeParser.kt [hunk 2]
```diff
@@ -62,7 +82,36 @@ abstract class AnimeParser : BaseParser() {
...
+            "vidstream.pro"               -> VidStreaming(server)
+            "mp4upload.com"               -> Mp4Upload(server)
+            "playtaku.net"                -> GogoCDN(server)
+            "alions.pro"                  -> ALions(server)
+            "awish.pro"                   -> AWish(server)
+            "dood.wf"                     -> DoodStream(server)
+            "ok.ru"                       -> OkRu(server)
+
+            //currently down so I can't add the extractors
+            "mcloud.to"         -> AniWave.Extractor(server) // https://mcloud.to/e/qp2405?t=4xnZC%2FwjB1MKyw%3D% I don't think the AniWave.Extractor() is a good solution because it's site dependent.
```
Comment (brahmkshatriya) [inline app/src/main/java/ani/saikou/parsers/AnimeParser.kt:103]: remove

File: app/src/main/java/ani/saikou/parsers/AnimeParser.kt
Comment (brahmkshatriya) [inline app/src/main/java/ani/saikou/parsers/AnimeParser.kt:1]: The reason why I didn't do this, is because of the unreliability of the domains changing constantly

Now imagine if some site changes their domain or keeps adding new domains every week, 
what would we do? we would add that domain here

now for every AnimeParser that uses the super method, this will have to make checks for those domains that are not useful to that parser. (ofc we could not add such domains here, but its something to keep in mind)
